test/framework/clustervalidator.go

```diff
@@ -0,0 +1,92 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type clusterValidation = func(client client.Client, config *cluster.Config) error
```

Comment:
You might want to consider passing the `cluster.Spec`
That's the one that will also include the `Bundles`

You'll find that to run certain validations, you will need the version of the components installed (eks-d version, cilium version, etc.). These versions are specified in the `Bundles`, so with `cluster.Config` you don't have them but with `cluster.Spec` you do. 

The other option would be to just pass both the `cluster.Config` and the `Bundles` (or probably better, `VersionsBundle`, since that once is already specific for one kubernetes version).

Reply:
Ah, thanks for the heads up!